x/concentrated-liquidity/incentives_test.go

```diff
@@ -3368,3 +3379,133 @@ func (s *KeeperTestSuite) TestFindUptimeIndex() {
+	}
+	defaultConcentratedAssets := sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(100)), sdk.NewCoin("bar", sdk.NewInt(100)))
+	defaultBalancerPoolParams := balancer.PoolParams{SwapFee: sdk.NewDec(0), ExitFee: sdk.NewDec(0)}
+	tests := map[string]struct {
+		existingClLiquidity sdk.Coins
+		balancerPoolAssets  []balancer.PoolAsset
+
+		canonicalBalancerPoolDoesNotExist bool
+
+		expectedError error
+	}{
+		"happy path: balancer and CL pool at same spot price": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets:  defaultBalancerAssets,
+		},
+		"same spot price, different total share amount": {
+			// 100 existing shares and 200 shares added from balancer
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets: []balancer.PoolAsset{
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("foo", sdk.NewInt(200))},
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("bar", sdk.NewInt(200))},
+			},
+		},
+		"different spot price between balancer and CL pools (excess asset0)": {
+			// 100 existing shares and 100 shares added from balancer. We expect only the even portion of
+			// the Balancer pool to be joined, with the remaining 50foo not qualifying.
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets: []balancer.PoolAsset{
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("foo", sdk.NewInt(150))},
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("bar", sdk.NewInt(100))},
+			},
+		},
+		"different spot price between balancer and CL pools (excess asset1)": {
+			// 100 existing shares and 100 shares added from balancer. We expect only the even portion of
+			// the Balancer pool to be joined, with the remaining 50bar not qualifying.
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets: []balancer.PoolAsset{
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("foo", sdk.NewInt(100))},
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("bar", sdk.NewInt(150))},
+			},
+		},
+		"canonical balancer pool ID exists but pool itself is not found": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity:               defaultConcentratedAssets,
+			balancerPoolAssets:                defaultBalancerAssets,
+			canonicalBalancerPoolDoesNotExist: true,
+
+			expectedError: gammtypes.PoolDoesNotExistError{PoolId: invalidPoolId},
+		},
+	}
+	for name, tc := range tests {
+		s.Run(name, func() {
+			// --- Setup test env ---
+
+			s.SetupTest()
+			clPool := s.PrepareCustomConcentratedPool(s.TestAccs[0], tc.existingClLiquidity[0].Denom, tc.existingClLiquidity[1].Denom, DefaultTickSpacing, DefaultExponentAtPriceOne, sdk.ZeroDec())
+
+			// Set up an existing full range position
+			initialLiquidity, _ := s.SetupPosition(clPool.GetId(), s.TestAccs[0], tc.existingClLiquidity[0], tc.existingClLiquidity[1], DefaultMinTick, DefaultMaxTick, s.Ctx.BlockTime())
```

Comment:
I believe this is nulling an err which we should be checking

Reply:
The second return value for `SetupPosition` is the position ID, not an error. Added a comment to clarify.